linkerd/app/outbound/src/metrics.rs

```diff
@@ -88,6 +89,11 @@ where
 
 impl PromMetrics {
     pub fn register(registry: &mut prom::Registry) -> Self {
+        let http_detect = crate::http::DetectMetricsFamilies::register(
+            // Scoped consistently with the inbound metrics.
+            registry.sub_registry_with_prefix("tcp_detect_http"),
```

Comment:
i haven't fully wrapped my head around why here we create a sub-registry with a `tcp_detect_http` prefix, but in the inbound metrics we create a `tcp_detect` sub-registry and an `http` sub-registry of _that_ sub-registry in two distinct steps.

i suspect that's due to differences in the inbound/outbound proxy components that i'm just not familiar with yet, but i found this a little confusing.

Reply:
Here's the thought process:

* on the inbound side, there is a distinct 'detect' stack that does both HTTP and TLS detection. when we build metrics for the detect stack we prefix the registry with 'tcp_detect' so that it can internally scope 'http' and (later?) 'tls'.
* on the outbound side, we don't have tls detection, nor a detect stack that works that way.

Reply:
ah! that makes sense, thank you.

would you mind recording some of this rationale in the comment here? `scoped consistently` otherwise might be a little bit cryptic for future readers